### PR TITLE
Remove environment concept from model

### LINKED ISSUE #3325: Remove environment indexes
**What would you like to be added**:

We still keep them for a while. They will be removed when the time comes.

**Why is this needed**:

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

Remove environment concept from:
- models: Application, Deployment.
- struct: ApplicationLiveState, PlanPreview, ApplicationInfo

**Which issue(s) this PR fixes**:

Fixes #3325

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/model/planpreview.proto [hunk 1]
```diff
@@ -44,10 +44,6 @@ message ApplicationPlanPreviewResult {
     ApplicationKind application_kind = 4 [(validate.rules).enum.defined_only = true];
     string application_directory = 5 [(validate.rules).string.min_len = 1];
 
-    string env_id = 6 [deprecated = true];
-    string env_name = 7 [deprecated = true];
-    string env_url = 8 [deprecated = true];
```
Comment (nghialv) [inline pkg/model/planpreview.proto:49]: Let's reserve those fields.
https://developers.google.com/protocol-buffers/docs/proto3#reserved

File: pkg/model/deployment.proto [hunk 2]
```diff
@@ -54,13 +54,12 @@ enum StageStatus {
...
 // - New commit was added and it mades a change on application configuration.
 // - User clicked on Sync button at application web page.
-// - Promoted from another enviroment.
+// - Triggered by another deployment.
 message Deployment {
     // The generated unique identifier.
     string id = 1 [(validate.rules).string.min_len = 1];
     string application_id = 2 [(validate.rules).string.min_len = 1];
     string application_name = 3 [(validate.rules).string.min_len = 1];
-    string env_id = 4 [deprecated = true];
```
Comment (nghialv) [inline pkg/model/deployment.proto:63]: Let's reserve its index number.
https://developers.google.com/protocol-buffers/docs/proto3#reserved

File: pkg/model/common.proto [hunk 3]
```diff
@@ -71,8 +71,6 @@ message ApplicationInfo {
     string config_filename = 7;
     string piped_id = 8 [(validate.rules).string.min_len = 1];
     string description = 9;
-    // This field will be no longer needed as labels can be an alternative.
-    string env_name = 14 [deprecated=true];
```
Comment (nghialv) [inline pkg/model/common.proto:75]: Here as well.

File: pkg/model/application_live_state.proto [hunk 4]
```diff
@@ -29,7 +29,6 @@ message ApplicationLiveStateSnapshot {
         OTHER = 2;
     }
     string application_id = 1 [(validate.rules).string.min_len = 1];
-    string env_id = 2 [deprecated = true];
```
Comment (nghialv) [inline pkg/model/application_live_state.proto:32]: Here as well. And other places.

**Review by nghialv — APPROVED** [review]
Nice clean-up.

**Review by knanao — APPROVED** [review]
Cool.

**Pull request merged**